omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,396 @@
+           
+            IdentifierPool identifierSource = new IdentifierPool();
+            Object name = updateBody.get("name");
+            Object description = updateBody.get("description");
+            Object batchSize = updateBody.get("batchSize");
+            Object minPoolSize = updateBody.get("minPoolSize");
+            Object sourceUuid = updateBody.get("sourceUuid");
+            Object sequential = updateBody.get("sequential");
+            Object refillWithScheduledTask = updateBody.get("refillWithScheduledTask");
+            
+            if (sourceUuid != null && StringUtils.isNotBlank(sourceUuid.toString())) {
+            
+                IdentifierSource poolIdentifierSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                if(poolIdentifierSource != null){
+                    identifierSource.setSource(poolIdentifierSource);
+                }else{
+                    errors.add("identifier source");
+                }
+            }
+            if (name != null && StringUtils.isNotBlank(name.toString())) {
+               identifierSource.setName(name.toString());  
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (batchSize != null && StringUtils.isNotBlank(batchSize.toString())) {
+                if(StringUtils.isNumeric(batchSize.toString())) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize.toString()));
+                }else {
+                    errors.add("batch size");
+                }
+            }
+            if (minPoolSize != null && StringUtils.isNotBlank(minPoolSize.toString())) {
+                if(StringUtils.isNumeric(minPoolSize.toString())) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize.toString()));
+                }else {
+                    errors.add("min pool size");
+                }
+            }
+            if (sequential != null && StringUtils.isNotBlank(sequential.toString())) {
+                if(parseBoolean(sequential.toString()) != null) {
+                    identifierSource.setSequential(parseBoolean(sequential.toString()));
+                }else {
+                    errors.add("sequential");
+                }
+            }
+            if (refillWithScheduledTask != null && StringUtils.isNotBlank(refillWithScheduledTask.toString())) {
+                if(parseBoolean(refillWithScheduledTask.toString()) != null) {
+                    identifierSource.setSequential(parseBoolean(refillWithScheduledTask.toString()));
+                }else {
+                    errors.add("refill");
+                }
+            }
+            if(errors.toString().length() > 2) {
```

Comment:
Does this mean you ignore the first two errors? Why > 2?

Reply:
fixed.